OCPBUGS-15440: fix CMO to apply console-plugin pod.spec config

sthaha opened the pull request:
The console-plugin's deployment configs that need to applied to the deployment.spec.template.spec weren't getting applied because the changes were made to the copy of the object instead of the original object itself. The tests didn't catch the issue because the pod assertions didn't validate if any pods were found for the combination of namespace and label-selector

This commit fixes both by refering (pointer) to the original spec instead of the copy and by fixing the assertions fail if it fails to find any pods.

simonpasquier reviewed (commented):
/lgtm

note for a follow-up: we can also add a unit test in the manifests package for testing all CMO options related to the monitoring console plugin.

jan--f commented:
/jira refresh

sthaha commented:
/retest-required

sthaha commented:
@simonpasquier @jan--f I have removed the assertion that  I added that to `AssertPodConfiguration` to check that at least one pod matches the selector passed. 
```
			if len(pods.Items) == 0 {
				return fmt.Errorf("%w - failed to find Pods in %s matching %s", err, namespace, labelSelector)
			}

```
due to the following unrelated test failures which may be another bug to fix

```
=== RUN   TestClusterMonitorOSMConfig/assert_pod_configuration_is_as_expected
    assertions.go:347: timed out waiting for the condition: %!w(<nil>) - failed to find Pods in openshift-monitoring matching k8s-app=openshift-state-metrics
--- FAIL: TestClusterMonitorOSMConfig (307.04s)

    --- PASS: TestClusterMonitorTelemeterClientConfig/test_the_telemeter-client_deployment_is_rolled_out (6.02s)
    --- FAIL: TestClusterMonitorTelemeterClientConfig/assert_pod_configuration_is_as_expected (300.01s)

=== RUN   TestClusterMonitorTelemeterClientConfig/assert_pod_configuration_is_as_expected
    assertions.go:347: timed out waiting for the condition: %!w(<nil>) - failed to find Pods in openshift-monitoring matching app.kubernetes.io/component=dashboard

```

sthaha commented:
Just a note - for `osm` below are the labels set 
```
app.kubernetes.io/component=exporter
app.kubernetes.io/managed-by=cluster-monitoring-operator
app.kubernetes.io/name=openshift-state-metrics
app.kubernetes.io/part-of=openshift-monitoring
pod-template-hash=6d87776c5f
```

sthaha commented:
/retest-required

sthaha commented:
https://github.com/openshift/cluster-monitoring-operator/pull/2028 should fix the issues found while testing this PR.

marioferh reviewed (commented):
/lgtm

jan--f commented:
/lgtm

simonpasquier commented:
/skip
/retest-required
